src/main/java/org/scijava/io/handle/DataHandle.java

```diff
@@ -514,18 +515,10 @@ default int readUnsignedByte() throws IOException {
 
 	@Override
 	default short readShort() throws IOException {
-		final int ch0;
-		final int ch1;
-		if (isBigEndian()) {
-			ch0 = read();
-			ch1 = read();
-		}
-		else {
-			ch1 = read();
-			ch0 = read();
-		}
-		if ((ch0 | ch1) < 0) throw new EOFException();
-		return (short) ((ch0 << 8) + (ch1 << 0));
+		final byte[] buf = new byte[2];
```

Comment:
This will destroy performance. We cannot allocate a new byte array with every short read. You could use an instance field buffer of sufficient size to avoid the issue.

Reply:
How can we cache this in the interface though? Should we require all DataHandle implementations to provide a buffer array which is used for conversions?